lib/google_search_data_viewer/keyword/google_search_parser.ex

```diff
@@ -0,0 +1,61 @@
+defmodule GoogleSearchDataViewer.Keyword.GoogleSearchParser do
+  @css_search_selectors %{
+    top_adwords: "#tads > .uEierd a.sVXRqc[href]",
+    top_non_adwords: ".MhgNwc a[href]",
+    non_adwords: ".yuRUbf a[href]",
+    bottom_adwords: "#bottomads .uEierd a.sVXRqc[href]"
+  }
+
+  def get_url_data_from_html(html) do
+    {_, parsed_html} = Floki.parse_document(html)
+
+    url_data =
+      []
+      |> prepend_top_adwords(parsed_html)
+      |> prepend_top_non_adwords(parsed_html)
+      |> prepend_non_adwords(parsed_html)
+      |> prepend_bottom_adwords(parsed_html)
+
+    url_data
+  end
+
+  defp prepend_top_adwords(url_data, html) do
+    Enum.map(
+      html
+      |> Floki.find(@css_search_selectors.top_adwords)
+      |> Floki.attribute("href"),
+      fn url -> %{url: url, is_adword: true, is_top_adword: true} end
+    ) ++
+      url_data
```

Comment:
```suggestion
      links = html
      |> Floki.find(@css_search_selectors.top_adwords)
      |> Floki.attribute("href")
      
    Enum.map(links,
      fn url -> %{url: url, is_adword: true, is_top_adword: true} end
    ) ++  url_data
```

After extract like this, you can see we can implement a private method to get link from the `html` to make the code more clean

Reply:
Yeah good. Will try to extract the common parts from those 4 functions into another function due to the repition between each - only the css class is different and the resulting boolean fields